src/components/FeatureViews/AdminPortal/AdminUtils.js

```diff
@@ -0,0 +1,173 @@
+import { isSunday } from "date-fns";
+
+export const roleColor = (string)=>{
+    switch(string){
+        case "administrator":
+            return "red";
+        case "parent":
+            return "#812CD7";
+        case "student":
+            return "#378BFF";
+        case "instructor":
+            return "#1FA18A"
+        case "receptionist":
+            return "#D04E3D"
+        default:
+            return "no match"
+    }
+}
+
+export const statusColor = (x)=> {
+    switch(x){
+        case "0":
+            return "red";
+        case "1":
+            return "red";
+        case "2":
+            return "yellow";
+        case "3":
+            return "yellow";
+    }
+}
+
+export const initials = (first, last) => {
+    if (first) {
+        let x = 0;
+        let firstI = first.charAt(0);
+        if (firstI.match(/[A-Z]/)){
+        let firstI = first.charAt(0)
+        let lastI = last.charAt(0)
+        return firstI + lastI
+        }else{
+            let firstI = first.charAt(1)
+            let lastI = last.charAt(0)
+            return firstI + lastI
+        }
+    }
+}
+
+export const emailCheck= (x) => {
+    if (x){
+        return x
+    }
+    else {
+        return "Email n/a"
+    }
+
+}
+
+export const phoneCheck = (y) => {
+    switch (y) {
+        case ("nan"):
+            return "Phone n/a";
+        case null:
+            return "Phone n/a"
+        default:
+            y= y.replace(/\D/g,'');
+            y = y.replace(/(\d{3})(\d{3})(\d{4})/, "($1)-$2-$3")
+            return y;
+    }
+}
+
+export const parseDate = (date) => {
+    let x = Date.parse(date);
+    let y = Date.parse(new Date());
+    let ago = (y-x)/1000;
+    if (isNaN(x)){
+        return "n/a"
+    }
+    else{
+        switch (ago) {
+            case (ago<60):
+                return `Updated ${Math.round(ago)} seconds ago`;
+            case (ago<3600):
+                return `Updated ${Math.round(ago/60)} minutes ago`;
+            case (ago<86400):
+                return `Updated ${Math.round(ago/3600)} hours ago`;
+            case (ago<172800):
+                return `Updated 1 day ago`;
+            default:
+                return `Updated ${Math.round(ago/86400)} days ago`;
```

Comment:
I don't believe you can use conditions in `switch` statements like this. Use if/else statements instead.